### ISSUE TITLE
Dropzone: handle multiple files

### ISSUE THREAD

**Issue by jmsche** [symfony/ux#7]
It would be nice if Dropzone could handle multiple files.

I was expecting it would as its parent type is the FileType, that allows setting `multiple` option to `true`.

**Comment by tgalopin** [comment]
Yep, that's a good idea! I'm not sure how to handle it properly yet but that's a good idea :)

**Comment by AxelBriche** [comment]
Probably like https://www.dropzonejs.com, with multiple query in ajax for upload each image and return a valid or error response.

**Comment by Lenawel** [comment]
is Multiple File allowed or not yet ?

**Comment by connorhu** [comment]
I plan to make multiple and chunked file uploads in one of my pilot project. At this moment I don't find any "official" direction about templating of UX modules (at this moment small prototyes). This is problem at my symfony collection type ux module too.

**Comment by yassinehamouten** [comment]
Hello,
I am having the same issue. I did a few things to make it work.
I will try to make it available to everyone as a package

**Comment by expace-dev** [comment]
Hello,

someone have the solution to do multiple upload?

**Comment by yassinehamouten** [comment]
Hello,

You can see how I make it work there https://github.com/yassinehamouten/ux-dropzone-multiple/tree/main/Resources. I tried to do it as a package, but stimulus does not call my controller. If someone has an idea...

I changed:
- in the form template I added "multiple" in the file input https://github.com/yassinehamouten/ux-dropzone-multiple/blob/3d282f520811ae720b0b194b7d28363eba6a1502/Resources/views/form_theme.html.twig#L6
- and in the controller.js https://github.com/yassinehamouten/ux-dropzone-multiple/blob/main/Resources/assets/src/controller.js

**Comment by yassinehamouten** [comment]
It's look like this if you take the css too 
![image](https://user-images.githubusercontent.com/22196574/141649407-ef887b61-0000-4643-b9ea-136cbe628164.png)

**Comment by JML75** [comment]
thank fou for your work, does it work if you use the DropzoneType:: class in your formType ?

**Comment by tgalopin** [comment]
@yassinehamouten this is interesting, I wasn't sure we could do it with the multiple attribute, good news! That would be a great PR if you wish :) !

**Comment by daFish** [comment]
@yassinehamouten Would it be ok for you when I create a PR with your changes?

**Comment by JML75** [comment]
hello

not sure what your are talking about.

best 

JML

Le 24 oct. 2022 à 11:13, Marcus Stöhr ***@***.***> a écrit :


@yassinehamouten <https://github.com/yassinehamouten> Would it be ok for you when I create a PR with your changes?

**Comment by CodingBDX** [comment]
i'm interesting, i try it but solution not work... what's the package to install dropzone-multiple? or just replace dropzone ux??

**Comment by weaverryan** [comment]
I think @yassinehamouten basically forked symfony/ux-dropzone and made some tweaks to make it work with multiple - https://github.com/symfony/ux/issues/7#issuecomment-968084788 - if it's that simple, it would be great if someone made a PR here to add that support :)

**Comment by CodingBDX** [comment]
i try to use, this https://github.com/daFish/ux/tree/feat/dropzone-support-multiple-files/src/Dropzone/Resources

but it's not work at all...same result, just one picture
@yassinehamouten i try it but it's old code for old version...

**Comment by daFish** [comment]
@CodingBDX I have not yet finished the port in the linked PR. But I just rebased my branch and it is now up-to-date.

**Comment by nzurita** [comment]
> @CodingBDX I have not yet finished the port in the linked PR. But I just rebased my branch and it is now up-to-date.

Does this mean multiple file upload will be available soon?

**Comment by daFish** [comment]
> > @CodingBDX I have not yet finished the port in the linked PR. But I just rebased my branch and it is now up-to-date.
> 
> Does this mean multiple file upload will be available soon?

I could certainly need some help with finishing it.

**Comment by igi-1w3r53n** [comment]
Is there an update on this topic? It seems not to work out of the box with the current version (2.7.1).

**Comment by kristjan-kure** [comment]
Hi! We need multiple file upload. Who can help with this? I understand [daFish](https://github.com/daFish) has been developing it, but we need it live? Who on the Symfony development team can help us to work with this?

**Comment by janklan** [comment]
> Hi! We need multiple file upload. Who can help with this? I understand [daFish](https://github.com/daFish) has been developing it, but we need it live? Who on the Symfony development team can help us to work with this?

Welcome to open source. What's your budget to speed things up? I'm sure @daFish and others would find time to work on work-related things more readily than passion hobbies :-)

**Comment by igi-1w3r53n** [comment]
I had a lot of trouble with this solution. I switched to uppy (https://uppy.io/) and it works good with symfony.

**Comment by daFish** [comment]
@janklan After some thoughts I'm not going to finish this. It is based upon the work of @yassinehamouten and I'd rather propose to go with uppy.io - which is already mentioned by @igi-1w3r53n.

**Comment by janklan** [comment]
> @janklan After some thoughts I'm not going to finish this.

Fair enough mate, I've been using Dropzone.js directly myself ever since I found this limitation - pretty much the same day I found this component.

For everyone else: all you need to receive multiple files is making sure your JS dropzone component (whichever it is) uploads files individually behind the scene, and then something along this trivial piece:

```php

if (0 === $request->files->count()) {
    throw new BadRequestException('At least one file is required.');
}

if ($request->files->count() > 1) {
    throw new BadRequestException('Uploading multiple files at once is not supported.');
}

$uploadedFile = $request->files->get('file');
```

I'm honestly not convinced there is a need for an UX component and a dedicated form field for this. What does it actually simplify?

**Comment by cmartin6** [comment]
Is there an update on this topic? I need an option to allow multiple file-uploads using this dropzone-component...

**Comment by kristjan-kure** [comment]
I don't understand the priorities. They should be:

- Common UI components - the essence of multiple image upload is to save data. This should have been built once Symfony UX was developed.
- Multiple select - we had to use https://tom-select.js.org/
- Modal - I don't see it on https://ux.symfony.com/.
- WYSIWYG editor - I don't see it on https://ux.symfony.com/.
- Gallery
- Date picker - I don't see it on https://ux.symfony.com/.

etc.

**Comment by janklan** [comment]
> I don't understand the priorities. They should be:

I find myself saying this more often than I like: welcome to open source. Feel free to contribute a solution to a real-world problem you're experiencing.

It's not easy, which is the reason why there are more takers than givers. If it helps you do your job and make a living, don't give advice how things should be, and give your time and skills instead :)

**Comment by kristjan-kure** [comment]
> > I don't understand the priorities. They should be:
> 
> I find myself saying this more often than I like: welcome to open source. Feel free to contribute a solution to a real-world problem you're experiencing.
> 
> It's not easy, which is the reason why there are more takers than givers. If it helps you do your job and make a living, don't give advice how things should be, and give your time and skills instead :)

Sorry, I am not a back-end developer, so I can't help build it, and I am not here to criticize. However, I can suggest a direction and discuss the need from my personal perspective.

With my experience of 20 years in web development, I believe the UX project requires a significant change in direction. One possible approach is to engage the community by allowing them to vote for UI elements. Those who have the time and expertise can contribute to building it. Votes will define the priorities. I do understand that you will need to figure out the message to send and motivate voters.

Here's an example: https://community.talk.hyvor.com/. This platform utilizes https://www.featurebase.app/ for feature voting.

**Comment by kristjan-kure** [comment]
The end goal here is to establish Symfony as a prominent PHP web application framework in the market?

**Comment by ytilotti** [comment]
@daFish had done most of the work. The problem is only on the preview because ux-dropzone manage already multiple files

**Comment by smnandre** [comment]
https://github.com/symfony/ux/pull/3769
